### PR TITLE
[UI Tests] Added "Products Search by SKU" e2e test

### PR DESCRIPTION
**Pull request by pachlava**
### Description
This PR adds another Products Search test, named `e2eRealApiProductsSearchBySKU`. This time search is done by SKU value. Besides the new test, there are no major changes, just moving some things around.

What the test does:
- Searches for a simple product by SKU
- Searches for two variable products that share a part of SKU
- Searches for exact variable product by SKU

### Testing instructions
- Make sure all tests run locally and on CI

### CODE REVIEW

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt [hunk 1]
```diff
@@ -69,26 +72,28 @@ class ProductsRealAPI : TestBase() {
...
+        priceDiscountedRaw = "2", sku = "CF-CPC"
+    )
+
+    private val productCappuccinoAlmondMedium = ProductData(
+        name = "Cappuccino", stockStatusRaw = "instock", priceDiscountedRaw = "3", sku = "CF-CPC-ALM-M"
+    )
+
+    private val productCappuccinoAlmondLarge = ProductData(
+        name = "Cappuccino", stockStatusRaw = "instock", priceDiscountedRaw = "4", sku = "CF-CPC-ALM-L"
+    )
```
Comment (jostnes) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt]: could be just me, but i find it easier to read in the previous version (each item in a line on its own)

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt [hunk 2]
```diff
@@ -112,4 +117,25 @@ class ProductsRealAPI : TestBase() {
...
+            .assertProductsCount(1)
+            // Search for variations sharing a part of SKU
+            .enterSearchTerm(productCappuccino.sku + "-ALM")
+            .assertProductCard(productCappuccinoAlmondMedium)
+            .assertProductCard(productCappuccinoAlmondLarge)
+            .assertProductsCount(2)
+            // Search for exact variation SKU
+            .enterSearchTerm(productCappuccinoAlmondLarge.sku)
+            .assertProductCard(productCappuccinoAlmondLarge)
+            .assertProductsCount(1)
```
Comment (jostnes) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt:149]: what is the difference between `Search for exact variation SKU` and `Search for a simple product SKU` ? both look to be looking for the exact SKU. i can see that one is an "extended" sku and the other is not but i'm wondering in terms of search, would that matter?

**Review by jostnes — APPROVED** [review]
Thanks for adding this too! Tested this locally and it's ✅ 

I added some comments, but again I don't think those are blocking (one is a question to understand why)

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt (hunk 1 above)
Comment (pachlava) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt]: Thanks @jostnes, I actually went for your suggestion with 2f33ea33fde4fbdb0915476a21f1cbe42128217e

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt (hunk 2 above)
Comment (pachlava) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/ProductsRealAPI.kt:149]: The difference is that product list, in non-search mode, does not display the individual variations, it displays the "parent" variable product only. `Search for exact variation SKU` means that we are searching for an individual variation by its SKU (not for the "parent" product), and expect to see it on screen, in that case.

**Comment by pachlava** [comment]
Thank you for the review @jostnes! 🙇 

I reformatted test data as you suggested and answered your question. Merging it now 🚀

**Pull request merged**